datafusion/functions-aggregate/src/count.rs

```diff
@@ -97,7 +97,10 @@ impl Default for Count {
 impl Count {
     pub fn new() -> Self {
         Self {
-            signature: Signature::variadic_any(Volatility::Immutable),
+            signature: Signature::one_of(
```

Comment:
This is necessary so that the planner doesn't initially fail the query. It shouldn't progress to that point, but perhaps there's a good way to add an error if `Count` does get called with empty args?

Reply:
It would probably be good to call that out with a comment:

```suggestion
            signature: Signature::one_of(
              // TypeSignature::Any(0) is required to handle `Count()` with no args
```

Maybe as a follow on PR we could look into making `TypeSignature::VariadicAny` accept zero arguments (which I certainly would have expected, given the name)

Another way might be to use https://docs.rs/datafusion/latest/datafusion/logical_expr/enum.TypeSignature.html#variant.UserDefined and then overriding 

https://docs.rs/datafusion/latest/datafusion/logical_expr/trait.ScalarUDFImpl.html#method.coerce_types

However that seems like it would be more complex than allowing VariadicAny to take 0 args

Reply:
Filed https://github.com/apache/datafusion/issues/11522